pkg/api/handlers/compat/images.go

```diff
@@ -179,6 +180,18 @@ func CreateImageFromSrc(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 	// fromSrc – Source to import. The value may be a URL from which the image can be retrieved or - to read the image from the request body. This parameter may only be used when importing an image.
+
+	platformSpecs := strings.Split(query.Platform, "/")
+
+	addOS := true
```

Comment:
Does this work.
addOS := len(platformSpecs) == 0 
addArch := len(platformSpecs) > 1

Reply:
I don't think you need addOS and you should just move this code down below where you are setting the OS and Arch.